middleware/auth/jwt/jwt.go

```diff
@@ -104,27 +104,23 @@ func Server(keyFunc jwt.Keyfunc, opts ...Option) middleware.Middleware {
 					tokenInfo, err = jwt.Parse(jwtToken, keyFunc)
 				}
 				if err != nil {
-					ve, ok := err.(*jwt.ValidationError)
-					if !ok {
-						return nil, errors.Unauthorized(reason, err.Error())
-					}
-					if ve.Errors&jwt.ValidationErrorMalformed != 0 {
-						return nil, ErrTokenInvalid
-					}
-					if ve.Errors&(jwt.ValidationErrorExpired|jwt.ValidationErrorNotValidYet) != 0 {
+					if errors.IsUnauthorized(err) || errors.Is(err, jwt.ErrTokenMalformed) ||
+						errors.Is(err, jwt.ErrTokenNotValidYet) || errors.Is(err, jwt.ErrTokenExpired) {
 						return nil, ErrTokenExpired
 					}
-					if ve.Inner != nil {
-						return nil, ve.Inner
-					}
 					return nil, ErrTokenParseFail
 				}
+
 				if !tokenInfo.Valid {
 					return nil, ErrTokenInvalid
 				}
-				if tokenInfo.Method != o.signingMethod {
+
+				if _, ok := tokenInfo.Method.(*jwt.SigningMethodHMAC); !ok {
```

Comment:
Why doesn't it support other signature methods?

Reply:
Sorry, I misunderstood the logic of this position and have resolved this issue on my next commit